src/backend/marsha/core/tests/test_api_video.py

```diff
@@ -2654,6 +2726,103 @@ def test_api_video_patch_detail_token_user_is_public(self):
+        video = factories.VideoFactory(allow_recording=True)
+        jwt_token = AccessToken()
+        jwt_token.payload["resource_id"] = str(video.id)
+        jwt_token.payload["roles"] = [random.choice(["instructor", "administrator"])]
+        jwt_token.payload["permissions"] = {"can_update": True}
+        data = {"allow_recording": False}
+        response = self.client.patch(
+            f"/api/videos/{video.id}/",
+            data,
+            HTTP_AUTHORIZATION=f"Bearer {jwt_token}",
+            content_type="application/json",
+        )
+        self.assertEqual(response.status_code, 200)
+        video.refresh_from_db()
+        self.assertFalse(video.allow_recording)
+
+    def test_api_video_patch_detail_token_user_estimated_duration(self):
+        """Instructors and administrators should be able to
+        patch the estimated_duration field of their video through the API."""
+        video = factories.VideoFactory(estimated_duration=None)
+        jwt_token = AccessToken()
+        jwt_token.payload["resource_id"] = str(video.id)
+        jwt_token.payload["roles"] = [random.choice(["instructor", "administrator"])]
+        jwt_token.payload["permissions"] = {"can_update": True}
+        data = {"estimated_duration": "1:30:00"}
+        response = self.client.patch(
+            f"/api/videos/{video.id}/",
+            data,
+            HTTP_AUTHORIZATION=f"Bearer {jwt_token}",
+            content_type="application/json",
+        )
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.json().get("estimated_duration"), "01:30:00")
+        video.refresh_from_db()
+        self.assertEqual(video.estimated_duration, timedelta(hours=1, minutes=30))
+
+    def test_api_video_patch_detail_token_user_estimated_duration_seconds(self):
+        """Instructors and administrators should be able to
+        patch the estimated_duration field of their video through the API
+        by sending only the duration value in seconds."""
+        video = factories.VideoFactory(estimated_duration=None)
+        jwt_token = AccessToken()
+        jwt_token.payload["resource_id"] = str(video.id)
+        jwt_token.payload["roles"] = [random.choice(["instructor", "administrator"])]
+        jwt_token.payload["permissions"] = {"can_update": True}
+        data = {"estimated_duration": "5400"}
+        response = self.client.patch(
+            f"/api/videos/{video.id}/",
+            data,
+            HTTP_AUTHORIZATION=f"Bearer {jwt_token}",
+            content_type="application/json",
+        )
+        self.assertEqual(response.status_code, 200)
+        video.refresh_from_db()
+        self.assertEqual(video.estimated_duration, timedelta(hours=1, minutes=30))
+
+    def test_api_video_patch_detail_token_user_has_chat(self):
+        """Instructors and administrators should be able to
+        patch the chat flag of their video through the API."""
+        video = factories.VideoFactory(has_chat=True)
```

Comment:
I am more comfortable with this approach as has_chat is True by default, just to confirm it with the assign gives no wondering of the default value.
```suggestion
        video = factories.VideoFactory()
        self.assertTrue(video.has_chat)
```

Reply:
I don't think tests should rely on default values.
The point in this test is to assert that `has_chat` update is allowed in this context.

Reply:
exactly, with self.assertTrue we are sure it is defined to True, but to give has_chat=True to the factory is pointless and makes you wonder of it's default value

Reply:
Maybe I wasn't clear… let my rephrase my point:
I don't think that any tests should fail if we decide to change a default value.

Reply:
> I don't think that any tests should fail if we decide to change a default value.

I think tests should fail if you change default values. But it's not really the purpose of this specific test.

Default values are tested in all other tests modified in this PR, if you change one default all related tests will fail.

In this specific test, testing the default value can be the "cherry on the cake". Both scenarios are tested. You assert the default value is the one expected and you assert that you can update its value.

Reply:
>Default values are tested in all other tests modified in this PR, if you change one default all related tests will fail.
you're right :facepalm: